lib/token.ts

```diff
@@ -771,16 +771,22 @@ function removeSearch(sdk) {
   }
 }
 
-function getOAuthParamsStrFromStorage() {
+function _getOAuthParamsStrFromStorage() {
   let oauthParamsStr;
   if (browserStorage.browserHasSessionStorage()) {
-    const storage = browserStorage.getSessionStorage();
-    oauthParamsStr = storage.getItem(REDIRECT_OAUTH_PARAMS_NAME);
-    storage.removeItem(REDIRECT_OAUTH_PARAMS_NAME);
-  } else {
+    oauthParamsStr = browserStorage.getSessionStorage().getItem(REDIRECT_OAUTH_PARAMS_NAME);  
+  }
+  if (!oauthParamsStr) {
+    // fallback to cookies to support legacy browsers, eg: IE/Edge, iphone 8
```

Comment:
The iOS issue is actually the other way, the **cookies** will fail because of of sameSite logic which is incompatible with the logic in iOS 13

Reply:
wondering why it failed for the last hard `sessionStorage` switch change?

Reply:
Because the logic was

```
if (browserHasSessionStorage()) {
  // read from session storage
} else {
  // read from cookie
}
```

It was the **else** that caused the break. Session storage is enabled, but the value was in the cookie.  We decided to favor cookie over session storage in that fix to also overcome same-tab concurrency (since authJS was only writing to sessionStorage), however multi-tab concurrency still exists.

With authJS writing to both values, there will be a potential single-tab concurrency issue, but we are fighting this in other ways. I do think it is correct to favor the value read from session storage first, to better defend against multi-tab concurrency.